src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs

```diff
@@ -344,87 +325,73 @@ private DriveInfo ReportCommonDiskUsage(string dirType, string directory)
             return returnStatus;
         }
 
-        private async System.Threading.Tasks.Task<bool> CleanupDirectoryAsync(string directory, int attempts = 0)
+        private async System.Threading.Tasks.Task<bool> CleanupDirectoryAsync(string directory, int attempts = 0, bool ignoreExceptions = false)
         {
             try
             {
-                if (Directory.Exists(directory))
+                // A protected directory is, for instance, the directory which is currently running the cleanup task.
+                // The cleanup task should never try to clean itself up
+                foreach (string protectedDirectory in protectedDirectories)
+                {
+                    if (protectedDirectory.Contains(directory))
```

Comment:
Regarding directory comparisons: what is the preferred alternative?

Regarding root directories: I considered including the vice-versa of this, but convinced myself it wasn't a valid scenario.  ie...

```
if (protectedDirectory.Contains(directory) || directory.Contains(protectedDirectory))
```